src/frontend/js/login.js

```diff
@@ -149,11 +217,38 @@ div:has(#submit_btn) {
     this.password = inputEl.value;
   }
 
+  checkLogin(e) {
+    if (e.key == "Enter") this.loginConfirm();
+  }
+
   setVisibility() {
-    this.pswVisibility = !this.pswVisibility
+    this.pswVisibility = !this.pswVisibility;
+  }
+
+  getElement(query) {
+    return this.renderRoot.querySelector(query) ?? null;
   }
 
   loginConfirm() {
+    if (
+      this.getElement("#username").value === "" &&
+      this.getElement("#password").value === ""
+    ) {
+      this.getElement("#usernameError").innerText = "*Inserisci uno username";
```

Comment:
anche qui non andrei a recuperare l'elemento tramite l'id ma userei un approccio stile Lit usando una variabile che contiene il contenuto da mostrare nell'elemento

Reply:
Creato delle reactive properties che aggiorno in caso di errore